### ISSUE TITLE
[WhatsApp] problems with creating groups

### ISSUE THREAD

**Issue by ghost** [loqui/im#712]
If i create a Whatsapp group with Loqui, a group with the name 'null'
![png-grafik 320 x 510 pixel - skaliert 78](https://cloud.githubusercontent.com/assets/4523264/7550830/242b3686-f671-11e4-8d70-2a18a4cac494.png)
is created and i can't access the group settings

```
TypeError: participantJid is null                         messenger.js:177:8
```

There seems to be also the possibility to invite contacts during group creation, but it does not work, too

```
TypeError: selected.indexOf is not a function             account.js:316:0
```

After a restart of Loqui the group with the name i created appears, and i could invite contacts successfully, but the group 'null' still exist.

Firefox OS v2.0 / Firefox OS Simulator v2.0 / Loqui IM v0.4.2b

**Comment by TitanNano** [comment]
yes I experience the same. But I could remove the null group.

**Comment by gargsms** [comment]
Is there already an issue for deleting members from a group? The groups are missing a lot of functionality in Loqui.
I have not used WhatsApp for the last 3 months now but now that my college is over, I will head straight into debugging Loqui. Hopefully.

**Comment by TitanNano** [comment]
@gargsms I don't thinks so. but I'll go into group enhancement soon!

**Comment by ghost** [comment]
@TitanNano Your right the null group could be removed, i didn't try it from the chat list before.

**Comment by cmeerw** [comment]
The "TypeError: selected.indexOf is not a function" still remains...

**Comment by TitanNano** [comment]
@cmeerw can you give me the exact line number this error occurs?  I already fixed this once but unfortunately lost the patch due to some mistake >.<

**Comment by cmeerw** [comment]
I get

`TypeError: selected.indexOf is not a function (account.js:338:22)`

when clicking on "+ Invite" in the "Create a group" screen.

And then I get the following error when actually creating the group:

`Error details: TypeError: "aParticipants.forEach is not a function" at modifyGroupParticipants@scripts/mozillahispano/coseme.js:7747:7
methodList.group_addParticipants@scripts/mozillahispano/coseme.js:8053:1
call@scripts/mozillahispano/coseme.js:6503:14
App.connectors.coseme/this.events.onGroupCreateSuccess@scripts/loqui/connectors/coseme.js:664:1
fireEvent/@scripts/mozillahispano/coseme.js:7458:9
fireEvent@scripts/mozillahispano/coseme.js:7456:5
parseGroupCreated@scripts/mozillahispano/coseme.js:7174:7
processNode.result@scripts/mozillahispano/coseme.js:6569:9
handleNode@scripts/mozillahispano/coseme.js:6650:9
_processTree@scripts/mozillahispano/coseme.js:5314:13`

**Comment by tibtab** [comment]
when I create a group and try to invite someone, I get this error:

```
SHOW mucInvite <li data-menu-onclick="mucInvite" data-icon="plus" data-l10n-id="Actioninvite" class="add icon plus"> undefined menu.js:297:4
TypeError: selected.indexOf is not a function account.js:347:22
```

and when I create the group and then invite somebody i get this:

```
Error invoking group_addParticipants with ["me-1443306489@g.us",{"length":0,"prevObject":{"0":{"jQuery111006933390864546198":56},"length":1,"prevObject":{"0":{},"length":1,"prevObject":{"0":{"sizzle-1443299879119":{"parentNode":[1307,107,true]}},"length":1,"prevObject":{"0":{"location":{"href":"app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}/index.html","origin":"app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}","protocol":"app:","username":"","password":"","host":"{0ebd1812-eda1-4c22-81f0-36e495ee7d04}","hostname":"{0ebd1812-eda1-4c22-81f0-36e495ee7d04}","port":"","pathname":"/index.html","search":"","hash":""},"avatar":{},"webL10n":{},"jQuery111006933390864546198":120},"context":{"location":{"href":"app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}/index.html","origin":"app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}","protocol":"app:","username":"","password":"","host":"{0ebd1812-eda1-4c22-81f0-36e495ee7d04}","hostname":"{0ebd1812-eda1-4c22-81f0-36e495ee7d04}","port":"","pathname":"/index.html","search":"","hash":""},"avatar":{},"webL10n":{},"jQuery111006933390864546198":120},"length":1},"context":{"location":{"href":"app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}/index.html","origin":"app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}","protocol":"app:","username":"","password":"","host":"{0ebd1812-eda1-4c22-81f0-36e495ee7d04}","hostname":"{0ebd1812-eda1-4c22-81f0-36e495ee7d04}","port":"","pathname":"/index.html","search":"","hash":""},"avatar":{},"webL10n":{},"jQuery111006933390864546198":120},"selector":"section#mucCreateForm article.form"},"context":{"location":{"href":"app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}/index.html","origin":"app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}","protocol":"app:","username":"","password":"","host":"{0ebd1812-eda1-4c22-81f0-36e495ee7d04}","hostname":"{0ebd1812-eda1-4c22-81f0-36e495ee7d04}","port":"","pathname":"/index.html","search":"","hash":""},"avatar":{},"webL10n":{},"jQuery111006933390864546198":120},"selector":"section#mucCreateForm article.form ul.listBox"},"context":{"location":{"href":"app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}/index.html","origin":"app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}","protocol":"app:","username":"","password":"","host":"{0ebd1812-eda1-4c22-81f0-36e495ee7d04}","hostname":"{0ebd1812-eda1-4c22-81f0-36e495ee7d04}","port":"","pathname":"/index.html","search":"","hash":""},"avatar":{},"webL10n":{},"jQuery111006933390864546198":120}},"context":{"location":{"href":"app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}/index.html","origin":"app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}","protocol":"app:","username":"","password":"","host":"{0ebd1812-eda1-4c22-81f0-36e495ee7d04}","hostname":"{0ebd1812-eda1-4c22-81f0-36e495ee7d04}","port":"","pathname":"/index.html","search":"","hash":""},"avatar":{},"webL10n":{},"jQuery111006933390864546198":120}}] ~ call@app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}/scripts/mozillahispano/coseme.js:6505:7 [yowsupAPI] coseme.js:3541:4
putMessage() coseme.js:3541
Logger.prototype._message() coseme.js:3483
Logger.prototype.error() coseme.js:3471
call() coseme.js:6505
App.connectors.coseme/this.events.onGroupCreateSuccess() coseme.js:706
fireEvent/<() coseme.js:7458
forEach() self-hosted:173
fireEvent() coseme.js:7456
parseGroupCreated() coseme.js:7176
processNode.result() coseme.js:6569
handleNode() coseme.js:6650
_processTree() coseme.js:5314

Error details: TypeError: "aParticipants.forEach is not a function" at modifyGroupParticipants@app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}/scripts/mozillahispano/coseme.js:7747:7 > methodList.group_addParticipants@app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}/scripts/mozillahispano/coseme.js:8053:1 > call@app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}/scripts/mozillahispano/coseme.js:6503:14 > App.connectors.coseme/this.events.onGroupCreateSuccess@app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}/scripts/loqui/connectors/coseme.js:706:1 > fireEvent/<@app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}/scripts/mozillahispano/coseme.js:7458:9 > fireEvent@app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}/scripts/mozillahispano/coseme.js:7456:5 > parseGroupCreated@app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}/scripts/mozillahispano/coseme.js:7176:7 > processNode.result@app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}/scripts/mozillahispano/coseme.js:6569:9 > handleNode@app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}/scripts/mozillahispano/coseme.js:6650:9 > _processTree@app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}/scripts/mozillahispano/coseme.js:5314:13 >  ~ call@app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}/scripts/mozillahispano/coseme.js:6506:7 [yowsupAPI] coseme.js:3541:4
putMessage() coseme.js:3541
Logger.prototype._message() coseme.js:3483
Logger.prototype.error() coseme.js:3471
call() coseme.js:6506
App.connectors.coseme/this.events.onGroupCreateSuccess() coseme.js:706
fireEvent/<() coseme.js:7458
forEach() self-hosted:173
fireEvent() coseme.js:7456
parseGroupCreated() coseme.js:7176
processNode.result() coseme.js:6569
handleNode() coseme.js:6650
_processTree() coseme.js:5314

FireEvent exception! TypeError: "aParticipants.forEach is not a function" at modifyGroupParticipants@app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}/scripts/mozillahispano/coseme.js:7747:7 > methodList.group_addParticipants@app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}/scripts/mozillahispano/coseme.js:8053:1 > call@app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}/scripts/mozillahispano/coseme.js:6503:14 > App.connectors.coseme/this.events.onGroupCreateSuccess@app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}/scripts/loqui/connectors/coseme.js:706:1 > fireEvent/<@app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}/scripts/mozillahispano/coseme.js:7458:9 > fireEvent@app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}/scripts/mozillahispano/coseme.js:7456:5 > parseGroupCreated@app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}/scripts/mozillahispano/coseme.js:7176:7 > processNode.result@app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}/scripts/mozillahispano/coseme.js:6569:9 > handleNode@app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}/scripts/mozillahispano/coseme.js:6650:9 > _processTree@app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}/scripts/mozillahispano/coseme.js:5314:13 >  ~ fireEvent/<@app://{0ebd1812-eda1-4c22-81f0-36e495ee7d04}/scripts/mozillahispano/coseme.js:7460:9 [ConnectionManager] coseme.js:3541:4
putMessage() coseme.js:3541
Logger.prototype._message() coseme.js:3483
Logger.prototype.error() coseme.js:3471
fireEvent/<() coseme.js:7460
forEach() self-hosted:173
fireEvent() coseme.js:7456
parseGroupCreated() coseme.js:7176
processNode.result() coseme.js:6569
handleNode() coseme.js:6650
_processTree() coseme.js:5314
```

**Comment by cmeerw** [comment]
This is supposed to be fixed with #869, closing
